lib/elixir/lib/option_parser.ex

```diff
@@ -492,4 +556,22 @@ defmodule OptionParser do
       :unknown
     end
   end
+
+  defp format_errors(errors) do
+    details = Enum.map_join(errors, ". ", &format_error(&1))
+    total = Enum.count(errors)
+    error_pluralize = pluralize(total, "error", "errors")
+    "You have #{total} #{error_pluralize}! #{details}"
+  end
+
+  defp format_error({option, nil}) do
+    "The option #{option} does not exist"
+  end
+
+  defp format_error({option, value}) do
+    "The option #{option} exists but it is of the wrong type: #{inspect value}"
```

Comment:
I agree with your suggestion, your message is more useful for the user and give a hint on how to solve the issue. I'll work on this and I'll let you know.